<html>
<body>
<head>
<script type="text/javascript" src="jquery.js"></script>
</head>
<TABLE style="position: absolute; left: 150px; width: 30">
<TR>
	<th style="background-color: #333399; border-width: 2px; color: #FFFFFF; font-size: 20px">Provenance of Microarray Experiments - for a Better Understanding of Experiment Results </th>
</TR>

<TR>
	<TD><BR><div style="border: 2px navy solid;""><I>Introduction</I><BR> Microarray experiments provide insights into the molecular mechanisms that lead to disease. The typical outcome of a microarray experiment is a list of differentially expressed genes that are potential targets for gene therapy and can lead to further studies of disease affected pathways. In these case study, we have collected 3 such genelists associated with Alzheimer's disease and represented the various levels of provenance RDF.</div> </TD>
</TR>
<TR>
	<td>
		<BR>
		<div style="background-color: #F7F7F7; font-size: 16px; text-align: left">
		<B>Quick Links:</B>
		<UL>
			<LI><a href="#queries">Demo Queries</a></Li>
			<LI><a href="http://biordfmicroarray.googlecode.com/hg/template_genelist.xml">Excel template to convert a genelist to RDF</a> (Right-click+Save as ...)</Li>
			<LI><a href="http://biordfmicroarray.googlecode.com/hg/genelist2rdf.html">Tool to convert genelists in Excel to RDF</a></Li>
		</UL>
		<BR>
		</div>
	</td>
</TR>
<TR>
	<TD><BR>
	<b>The Data Model:</b>
	<span>(<a href='#' onClick='javascript:$("#data_model").hide()'>hide</a>/<a href='#' onClick='javascript:$("#data_model").show()'>show</a>)</span>
	<div id='data_model'>
	 The data model to describe microarray experiments can be separated in four provenance levels. Each level entails different subsets of information: the laboratory where the experiment was performed and the reference where the results were published are part of the institutional level; the samples used in the experiment, the list of differentially expressed genes and the disease being studied are part of the experimental context level; the statistical and significance analysis tools and results are part of the statistical and significance analysis level and the metadata regarding the dataset itself is part of the dataset description level. <BR>
	
	<IMG SRC="provenance_levels2.png" WIDTH="1000" HEIGHT="600" BORDER="0" ALT="">
	</div>
	</TD>
</TR>
<TR>
		<TD style="color: #330066 font-size: 18px"><BR><B><a name="queries"></a>Example Queries:</B> <I>(Hint - click on the query to see it in the textarea)</I><BR>
		<form id="sparql-form" enctype="application/x-www-form-urlencoded" method="post" action="http://ibl.mdanderson.org/~mhdeus/sparql_federation/endpoint.php?" Target="sparql_result">
		
		
		<select id="query_select" onClick="fillTheTextArea();cacheOrQuery(this);" size="7">
		
				
<option id="Q0" value="
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> 
PREFIX dct: <http://purl.org/dc/terms/> 
PREFIX : <http://purl.org/net/biordfmicroarray/ns#>
PREFIX neurolex: <http://neurolex.org/wiki/Category:> 
PREFIX doid: <http://purl.org/obo/owl/DOID#> 
PREFIX magetab: <http://magetab-om.sourceforge.net/>

SELECT DISTINCT ?microarray_experiment ?title ?pubmedID WHERE 
{ 
   ?microarray_experiment	rdf:type	:microarray_experiment .
   ?experimentSet	dct:isPartOf   ?microarray_experiment .

   ?experimentSet	:has_input_value	?sampleList .
   ?sampleList	:derives_from_region	?brainRegion FILTER (?brainRegion = neurolex:Entorhinal_cortex ) .
   ?sampleList	:patients_have_disease	?alzheimers FILTER (?alzheimers = doid:DOID_10652 ) .

   ?microarray_experiment dct:isReferencedBy ?experimentRef .
   ?experimentRef magetab:title ?title .
   ?experimentRef :pubmed_link ?pubmedID .
}"
>
Q0. What microarray experiments analyze samples taken from the entorhinal cortex region of Alzheimer's patients?
</option>
<option  value="PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> 
PREFIX dct: <http://purl.org/dc/terms/> 
PREFIX : <http://purl.org/net/biordfmicroarray/ns#>
PREFIX neurolex: <http://neurolex.org/wiki/Category:> 
PREFIX doid: <http://purl.org/obo/owl/DOID#> 
PREFIX stat: <http://purl.org/net/biordfmicroarray/stat#>
PREFIX magetab: <http://magetab-om.sourceforge.net/>

SELECT DISTINCT ?normSoftware ?normSoftwareLabel ?microarray_experiment ?title WHERE 
{ 
   ?microarray_experiment	rdf:type	:microarray_experiment .
   ?experimentSet	dct:isPartOf   ?microarray_experiment .
   ?experimentSet	:has_input_value	?sampleList .
   ?sampleList	:derives_from_region	?brainRegion FILTER (?brainRegion = neurolex:Entorhinal_cortex ) .
   ?sampleList	:patients_have_disease	?alzheimers FILTER (?alzheimers = doid:DOID_10652 ) .

   ?microarray_experiment dct:isReferencedBy ?experimentRef .
   ?experimentRef magetab:title ?title .
   ?experimentRef :pubmed_link ?pubmedID .

   OPTIONAL { 
   ?microarray_experiment	stat:normalization_software	?normSoftware . 
   	?normSoftware rdfs:label ?normSoftwareLabel .
   }

}
"
>
Q1. What was the data normalization algorithm used in both studies that analyze gene expression in the entorhinal cortex region of AD patients?
</option>
<option value='PREFIX : <http://purl.org/net/biordfmicroarray/ns#>
PREFIX neurolex: <http://neurolex.org/wiki/Category:> 
PREFIX doid: <http://purl.org/obo/owl/DOID#> 
PREFIX stat: <http://purl.org/net/biordfmicroarray/stat#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX dct: <http://purl.org/dc/terms/> 
SELECT DISTINCT ?geneLabel ?foldChangeValue ?pval ?microarray_experiment WHERE 
{ 
  ?microarray_experiment	rdf:type	:microarray_experiment .
  ?experimentSet dct:isPartOf ?microarray_experiment	.
  ?experimentSet	:has_input_value	?sampleList .
  ?sampleList	:derives_from_region	?brainRegion FILTER (?brainRegion = neurolex:Entorhinal_cortex ) .
  ?sampleList	:patients_have_disease	?alzheimers FILTER (?alzheimers = doid:DOID_10652 ) .
   
  ?experimentSet	:differentially_expressed_gene	 ?gene .
  ?gene	rdfs:label ?geneLabel .
  ?experimentSet	:has_ouput_value ?foldChange .
  ?foldChange	rdf:type	:fold_change .
  ?foldChange	rdf:value	?foldChangeValue .
  ?foldChange	stat:p_value	?pval  .
  FILTER (?foldChangeValue > 0)
  FILTER (?pval < 0.001)
}
'
>
Q2: What genes are overexpressed in the entorhinal cortex region in the context of Alzheimer's and what is their expression fold change and associated p-value?
</option>
<option value="
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> 
PREFIX dct: <http://purl.org/dc/terms/> 
PREFIX : <http://purl.org/net/biordfmicroarray/ns#>
PREFIX neurolex: <http://neurolex.org/wiki/Category:> 
PREFIX doid: <http://purl.org/obo/owl/DOID#> 
PREFIX stat: <http://purl.org/net/biordfmicroarray/stat#>
SELECT DISTINCT ?geneLabel WHERE 
{ 
  
  ?ECgene :expressed_in	?ECsampleList .
  ?ECgene rdfs:label ?geneLabel .
 
  ?HIPgene :expressed_in	?HIPsampleList .
  ?HIPgene rdfs:label ?geneLabel . 
  
  ?ECsampleList :derives_from_region	neurolex:Entorhinal_cortex .
  ?HIPsampleList :derives_from_region	neurolex:Hippocampus .
  
}"
>
Q3: Are there any genes that are expressed differently in two different brain regions (such as in hippocampus and entorhinal cortex)?
</option>
<option value='PREFIX diseasome: <http://www4.wiwiss.fu-berlin.de/diseasome/resource/diseasome/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX stat: <http://purl.org/net/biordfmicroarray/stat#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX void: <http://rdfs.org/ns/void#>
PREFIX dct: <http://purl.org/dc/terms/>
PREFIX biordf: <http://purl.org/net/biordfmicroarray/ns#> 
PREFIX neurolex: <http://neurolex.org/wiki/Category:> 
PREFIX doid: <http://purl.org/obo/owl/DOID#> 

SELECT DISTINCT  ?diseaseName ?geneLabel ?geneName WHERE {

	#Retrieve a list of overexpressed genes in the entorhinal cortex of AD patients
	{
		?experimentSet	dct:isPartOf	?microarray_experiment	;
				biordf:has_input_value	?sampleList ;
				biordf:differentially_expressed_gene	 ?gene ;
				biordf:has_ouput_value ?foldChange .

		?sampleList	biordf:derives_from_region	?brainRegion ;
			biordf:patients_have_disease	?alzheimers .

		?gene	rdfs:label ?geneLabel ;
			biordf:name	   ?geneName .

		?foldChange	rdf:value	?foldChangeValue ;
			stat:p_value ?pval .

		#Apply filters to constrain the amount of results
		FILTER (xsd:float(?foldChangeValue) > 0)
		FILTER (xsd:float(?pval) < 0.001 )
		FILTER (?brainRegion = neurolex:Entorhinal_cortex ) 
		FILTER (?alzheimers = doid:DOID_10652 )
	}

	#Find most recently updated SPARQL endpoint that contains information about genes and diseases.
	
	{
		?source	rdf:type	void:Dataset ;
		void:sparqlEndpoint	?srvc ;
		dct:issued		?issued  ;
		dct:subject		diseasome:diseases ;
		dct:subject		diseasome:genes .

		OPTIONAL {
		?source1	rdf:type	void:Dataset ;
		void:sparqlEndpoint	?srvc2 ;
		dct:issued	?issued2 ;
		dct:subject	diseasome:diseases ; 
		dct:subject	diseasome:genes . 
		FILTER (?issued2 > ?issued) 
		}
		FILTER (!BOUND(?srvc2))
	}

	#Get associated diseases from most recently updated Diseasome server.
	SERVICE ?srvc {
		?diseasomeGene rdfs:label ?geneLabel .
		?disease diseasome:associatedGene ?diseasomeGene.
		?disease rdfs:label ?diseaseName .
	}
}
'
>
Q4: What other diseases may be associated with the same genes found to affect AD? <I>(Federated query)</I>
</option>
<option value="
PREFIX : <http://purl.org/net/biordfmicroarray/ns#> 
PREFIX diseasome: <http://www4.wiwiss.fu-berlin.de/diseasome/resource/diseasome/>
PREFIX ex: <http://purl.org/net/biordfmicroarray/ns#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX stat: <http://purl.org/net/biordfmicroarray/stat#>
PREFIX drugbk: <http://www4.wiwiss.fu-berlin.de/drugbank/resource/drugbank/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> 
PREFIX dct: <http://purl.org/dc/terms/> 

SELECT ?drugName ?OverExpressedGenLabel ?diseaseTarget WHERE {
   {
   ?microarray_experiment_set dct:isPartOf  ?microarray_experiment ;
							:differentially_expressed_gene	?OverExpressedGene ;
							:has_ouput_value ?foldChange .

   
   ?OverExpressedGene rdfs:label ?OverExpressedGenLabel .

  
   ?foldChange rdf:value ?foldChangeVal ;
				stat:p_value ?pval  .

   FILTER (xsd:float(?foldChangeVal) > 2) 
   FILTER (xsd:float(?pval) < 0.01 )
   }
 
  SERVICE <http://hcls.deri.org/sparql> {
        ?targetID drugbk:geneName ?OverExpressedGenLabel .
        ?drug drugbk:target ?targetID  .
        ?drug rdfs:label ?drugName .
        ?drug drugbk:mechanismOfAction ?actionMechanism .
        ?drug drugbk:possibleDiseaseTarget ?disease .
        ?disease rdfs:label ?diseaseTarget .
  }
}"
>
Q5: What drugs are known that affect the same overexpressed gene products and what are their target diseases?<I>(Federated query)</I>
</option>
<option value='PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> 
PREFIX dct: <http://purl.org/dc/terms/> 
PREFIX : <http://purl.org/net/biordfmicroarray/ns#>
PREFIX neurolex: <http://neurolex.org/wiki/Category:> 
PREFIX doid: <http://purl.org/obo/owl/DOID#> 
SELECT DISTINCT ?geneLabel ?geneName WHERE 
{ 
   ?microarray_experiment	rdf:type	:microarray_experiment ;
							dct:isReferencedBy ?reference .
   
   ?reference :experiment_laboratory	?lab .

   ?experimentSet	dct:isPartOf   ?microarray_experiment ;
					:has_input_value	?sampleList ;
					:differentially_expressed_gene ?gene .

   ?sampleList	:derives_from_region	?brainRegion FILTER (?brainRegion = neurolex:Entorhinal_cortex ) .
    ?sampleList	:patients_have_disease	?alzheimers FILTER (?alzheimers = doid:DOID_10652 ) .
   
   ?gene rdfs:label ?geneLabel ;
		 :name ?geneName .
	
	
	FILTER regex(?lab, "Translational Genomics Research Institute", "i")

}'
>
Q6: Select all the genes determined to be differentially expressed in the neurolex:Entorhinal_cortex. Use only experiments performed at the Translational Genomics Research Institute
</option>

</select>

	</TD>
</TR>
<TR>
	<TD><textarea id="query" name="query" rows="20" cols="128"></textarea></TD>
</TR>
<TR>
	<TD>
	<I>Output format (if supported by query type):</I>
	<select id="output" name="output">
              <option value="" >default</option>

              <option value="xml" >XML</option>
              <option value="json" >JSON</option>
              <option value="plain" >Plain</option>
              <option value="php_ser" >Serialized PHP</option>
              <option value="turtle" >Turtle</option>
              <option value="rdfxml" >RDF/XML</option>

              <option value="infos" >Query Structure</option>
              
              <option value="htmltab"  SELECTED >HTML Table</option>
              <option value="tsv" >TSV</option>
	</select>
	<span id="button_container">	
		<INPUT TYPE="submit" VALUE="Send Query" name="query_submited" id="query_submited" style='display:inline'>
		<INPUT TYPE="button" VALUE="See results" name="cached_query" id="cached_query" style='display: none'>
	</span>
	</FORM>
	<INPUT TYPE="button" VALUE="Clear" onClick="javascript:document.getElementById('query').innerHTML=''">
	</FORM>
	</TD>
</TR>
<TR>
	<TD>
	<BR><BR><BR><BR>
	</TD>
</TR>

</TABLE>
<script type="text/javascript">
function fillTheTextArea() {
	var browserName=navigator.appName; 
	var exQ = document.getElementById('query_select').value;
	document.getElementById('query').textContent=exQ;
	 if (browserName=="Microsoft Internet Explorer"){
		document.getElementById('query').value=exQ;
	}
	
	return false;
}
function cacheOrQuery(obj) {
	//if query is federated, change the button to see results; otherwise run the form
		var ch = obj.children;
		var sel = false;
		for (var i=0; i<ch.length; i++) {
			if(ch[i].selected){
				var sel = i;
			}
		}
		if(sel==4 || sel==5){
			
			$('#cached_query').click( function () {
				window.open("result_fedquery_q"+sel+".html","sparql_result");
			});

			$('#query').change(function () {
				alert('This is a federated query, editing this query will not alter the results in this interface. However, you can execute federated queries by using the tool available at http://sourceforge.net/projects/swobjects/files/. To reset changes click on the query again.');
			});
			
			$('#query_submited').hide();
			$('#cached_query').show();
			
		}
		else {
			$('#query_submited').show();
			$('#cached_query').hide();
			$('#query').removeAttr('onchange');
			
		}
	
}

function alterButton(q) {
	//this is necessary because for federated queries my endpoint does not work... Eric's does
	var formButton = document.getElementById('query_submited');
	switch (q) {
	case 0:
		document.getElementById('query_submited').disabled=false;
		document.getElementById('query_submited').setAttribute('type', 'submit');
		document.getElementById('sparql-form').setAttribute("action",'http://ibl.mdanderson.org/~mhdeus/sparql_federation/endpoint.php?');
	break;
	case 1:
		document.getElementById('query_submited').disabled=false;
		document.getElementById('query_submited').setAttribute('type', 'submit');
		document.getElementById('sparql-form').action='http://ibl.mdanderson.org/~mhdeus/sparql_federation/endpoint.php?';
	break;
	case 2:
		document.getElementById('query_submited').disabled=false;
		document.getElementById('query_submited').setAttribute('type', 'submit');
		document.getElementById('sparql-form').action='http://ibl.mdanderson.org/~mhdeus/sparql_federation/endpoint.php?';
	break;
	case 3:
		document.getElementById('query_submited').disabled=false;
		document.getElementById('query_submited').setAttribute('type', 'submit');
		document.getElementById('sparql-form').action='http://ibl.mdanderson.org/~mhdeus/sparql_federation/endpoint.php?';
	break;
	case 4:
		document.getElementById('query_submited').disabled=false;
		document.getElementById('query_submited').setAttribute('type', 'button');
		
		if( formButton.attachEvent ){
		   formButton.attachEvent('onclick', 'window.open("result_fedquery_q4.html","sparql_result")');
		} else {
		   formButton.setAttribute('onclick', 'window.open("result_fedquery_q4.html","sparql_result")'); 
		}
		
	break;
	case 5:
		document.getElementById('query_submited').disabled=false;
		document.getElementById('query_submited').setAttribute('type', 'button');
		if( formButton.attachEvent ){
		   formButton.attachEvent('onclick', 'window.open("result_fedquery_q5.html", ,"sparql_result")');
		} else {
		   formButton.setAttribute('onclick', 'window.open("result_fedquery_q5.html","sparql_result")'); 
		}
		//document.getElementById('sparql-form').setAttribute("action", 'result_fedquery_q5.html');
	break;
	case 6:
		document.getElementById('query_submited').disabled=false;
		document.getElementById('query_submited').setAttribute('type', 'submit');
		document.getElementById('sparql-form').action='http://ibl.mdanderson.org/~mhdeus/sparql_federation/endpoint.php?';
	break;
	case 7:
	    document.getElementById('query_submited').disabled=false;
		document.getElementById('query_submited').setAttribute('type', 'submit');
		document.getElementById('sparql-form').action='http://ibl.mdanderson.org/~mhdeus/sparql_federation/endpoint.php?';
	break;
	
	alert(document.getElementById('sparql-form').action);
	}
	
}
</script>

</body>